### ISSUE TITLE
Second-order gradient calculated in forward-mode for API `tf.linalg.det` is inaccurate relative to that in reverse-mode

### ISSUE THREAD

**Issue by eeDigitalSeal** [tensorflow/tensorflow#57430]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

tf 2.9

### Custom Code

Yes

### OS Platform and Distribution

Linux Ubuntu 20.04

### Python version

3.8

### Current Behaviour?

```shell
Second-order gradient calculated in forward-mode AD is different from that in reverse-mode. And the error reached 1e-1 which is not acceptable.
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf

input = tf.constant([[0.41278124, 0.472762  ],[0.5486289,0.8008381 ]],dtype=tf.float32)

with tf.GradientTape(persistent=True) as g2:
    with tf.GradientTape(persistent=True) as g1:
        g1.watch(input)
        g2.watch(input)
        res = tf.linalg.det(input)
    grad = g1.gradient(res, input)
grad2nd = g2.gradient(grad, input)

grad2nd_fwd = []
for i in range(tf.size(input)):
    hessian_col = []
    for j in range(tf.size(input)):
        tangent_inner = tf.reshape(tf.one_hot(i, tf.size(input), dtype=input.dtype), shape=input.shape)
        tangent_outer = tf.reshape(tf.one_hot(j, tf.size(input), dtype=input.dtype), shape=input.shape)
        with tf.autodiff.ForwardAccumulator(input, tangent_outer) as acc_outer:
            with tf.autodiff.ForwardAccumulator(input, tangent_inner) as acc_inner:
                res = tf.reduce_sum(tf.linalg.det(input))
            jvp = acc_inner.jvp(res)
        hvp = acc_outer.jvp(jvp)
        hessian_col.append(hvp)
    grad2nd_fwd.append(tf.reduce_sum(hessian_col))
grad2nd_fwd = tf.reshape(grad2nd_fwd, shape=input.shape)
print("reverse-mode\n",grad2nd)
print("forward-mode\n",grad2nd_fwd)
```

### Relevant log output

```shell
reverse-mode
 tf.Tensor(
[[ 1.0000001  -1.2048287 ]
 [-0.79517186  1.0000002 ]], shape=(2, 2), dtype=float32)
forward-mode
 tf.Tensor(
[[ 1.        -1.000001 ]
 [-1.0000002  1.       ]], shape=(2, 2), dtype=float32)
```
</details>

**Comment by sushreebarsa** [comment]
@eeDigitalSeal  I tried to reproduce the issue in TF2.9, tf-nightly(2.11.0-dev20220828),  and I faced different outputs. Could you please check this [gist](https://colab.research.google.com/gist/sushreebarsa/d1ce6beaa11866c3a4f0fb4d0563ac9b/untitled102.ipynb) and confirm the same?
Thank you!

**Comment by eeDigitalSeal** [comment]
@sushreebarsa Seems that this bug only appears in TF2.9.1 and already fixed in nightly. Thanks for pointing this out and I'll close this issue soon.
